src/main/java/org/dice_research/opal/civet/metrics/Dateformat2.java

```diff
@@ -0,0 +1,238 @@
+					continue;
+				}
+			}
+
+			for (String string : yearFormat) {
+				SimpleDateFormat Format = new SimpleDateFormat(string);
+				try {
+					Format.parse(issued.trim());
+					return 2;
+
+				} catch (ParseException pe) {
+					continue;
+				}
+			}
+		}
+		return 1;
+	}
+
+	public static int checkRange(String[] range) {
+		List<String> dateTimeFormat = new ArrayList<String>();
+		dateTimeFormat.add("YYYY-MM-DD hh:mm:s");
+		dateTimeFormat.add("dd.mm.yyyy hh:mm:s");
+		dateTimeFormat.add("dd-mm-yyyy hh:mm:s");
+		dateTimeFormat.add("dd/mm/yyyy hh:mm:s");
+		dateTimeFormat.add("YYYY.MM.DD hh:mm:s");
+		dateTimeFormat.add("YYYY/MM/DD hh:mm:s");
+
+		List<String> dateFormat = new ArrayList<String>();
+		dateFormat.add("YYYY-MM-DD");
+		dateFormat.add("dd.mm.yyyy");
+		dateFormat.add("YYYY/MM/DD");
+		dateFormat.add("YYYY.MM.DD");
+		dateFormat.add("dd-mm-yyyy");
+		dateFormat.add("dd/mm/yyyy");
+
+		List<String> yearFormat = new ArrayList<String>();
+		yearFormat.add("YYYY");
+
+		String fromDate = "", toDate = "";
+		if (range.length == 2) {
+			fromDate = range[0].trim();
+			toDate = range[1].trim();
+		}
+		if (fromDate.contains("GMT") && fromDate.split(" ").length == 6 && toDate.contains("GMT")
+				&& toDate.split(" ").length == 6)
+			return 4;
+		for (String string : dateTimeFormat) {
+			SimpleDateFormat Format = new SimpleDateFormat(string);
+			try {
+				Format.parse(fromDate);
+				Format.parse(toDate);
+				return 5;
+
+			} catch (ParseException pe) {
+				continue;
+			}
+		}
+
+		for (String string : dateFormat) {
+			SimpleDateFormat Format = new SimpleDateFormat(string);
+			try {
+				Format.parse(fromDate);
+				Format.parse(toDate);
+				return 4;
+
+			} catch (ParseException pe) {
+				continue;
+			}
+		}
+
+		for (String string : yearFormat) {
+			SimpleDateFormat Format = new SimpleDateFormat(string);
+			try {
+				Format.parse(fromDate);
+				Format.parse(toDate);
+				return 3;
+
+			} catch (ParseException pe) {
+				continue;
+			}
+		}
+		return 2;
+	}
+
+	private static final Logger logger = LogManager.getLogger();
+	private static final String descriptions = "if the date is in the format: DD/MM/YYYY hh:mm:s GMT - DD/MM/YYYY hh:mm:s GMT then give 5 stars"
+			+ "if the date is in the format: YYYY.MM.DD hh:mm:s GMT then give 4 stars"
+			+ "if the date is in the format: YYYY.MM.DD then give 3 stars"
+			+ "if the date is in the format: YYYY then give 2 stars" + "if the date property is empty then give 1 star";
+
+	@Override
+	public Integer compute(Model model, String datasetUri) throws Exception {
+		int result = 1;
+		logger.info("Processing dataset " + datasetUri);
+		Resource dataset = ResourceFactory.createResource(datasetUri);
+		StmtIterator datasetIterator = model.listStatements(new SimpleSelector(dataset, RDF.type, DCAT.Dataset));
+		if (datasetIterator.hasNext()) {
+			Statement dataSetSentence = datasetIterator.nextStatement();
+			Resource dataSet = dataSetSentence.getSubject();
```

Comment:
Line 189 - 192 is not required? Why are we iterating through datasets when we are always passing only 1 dataset. These 3 lines can be removed.